docs/root/configuration/http/cluster_specifier/lua.rst

```diff
@@ -0,0 +1,86 @@
+.. _config_http_cluster_specifier_lua:
+
+Lua cluster specifier
+=====================
+
+Overview
+--------
+
+The HTTP Lua cluster specifier allows `Lua <https://www.lua.org/>`_ scripts to select router cluster
+during the request flows. `LuaJIT <https://luajit.org/>`_ is used as the runtime. Because of this, the
+supported Lua version is mostly 5.1 with some 5.2 features. See the `LuaJIT documentation
+<https://luajit.org/extensions.html>`_ for more details.
+
+Configuration
+-------------
+
+* This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.router.cluster_specifiers.lua.v3.LuaConfig``.
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.router.cluster_specifiers.lua.v3.LuaConfig>`
+
+A simple example of configuring Lua cluster specifier is as follow:
+
+.. code-block:: yaml
+
+  name: envoy.router.cluster_specifier_plugin.lua
+  typed_config:
+    "@type": type.googleapis.com/envoy.extensions.router.cluster_specifiers.lua.v3.LuaConfig
+    source_code:
+      inline_string: |
+        function envoy_on_route(route_handle)
+          local header_value = route_handle:headers():get("header_key")
+          if header_value == "fake" then
+            return "fake_service"
+          end
+          return "web_service"
+        end
+    default_cluster: web_service
```

Comment:
It would be better to provide a complete route example here.

Reply:
done, thanks.